stack-previous.yaml

```diff
@@ -1 +0,0 @@
-resolver: nightly-2018-09-28
```

Comment:
Why?

Reply:
Why we need this `stack-previous.yaml` at all? I thought it was just a rudiment.

Reply:
Now I see that test is failed which in case using this `stack-previous.yaml` config. What purpose of this config? Could you elaborate on that? I think we should add an explanation for it to the README.

Reply:
stack-previous is for previous version of GHC

Reply:
@dzhus Relative to which one? Current in `stack.yaml`? So if `stack.yaml` has 8.6.x then `stack-previous.yaml` supposed to have 8.4.x? Why using nightly snapshot then instead of stable one? I'll add _lts-12.x_ resolver for `stack-previous.yaml`.

Reply:
I don't remember where I stole it from, it can definitely be seen in the wild (https://github.com/fpco/unliftio for example, but I can't really be arsed to rename files too) – this is just in case someone has to use previous GHC version.

Reply:
I don't use LTS resolvers because they offer little value and it's easier to see how old nightlies are from just the name of the nightly.

Reply:
1. Why do they offer little value comparing to nightlies? It's more clear by my opinion to which version of GHC it is attached just looking at major version of LTS, and it's more important to now major version of LTS which a lib is compatible with;
2. I don't see any benefits of knowing when a nightly snapshot have been published, but which LTS major version is more informative, since you could easily remember which version of GHC it is attached to.

Reply:
> benefits of knowing when a nightly snapshot have been published

Its freshness indicates how out-of-date is your package compared to a (somewhat stable subset of) the ecosystem. LTS number is just a number, it doesn't tell me anything, you shouldn't have to remember a 1-to-many mapping. I can see value in LTS if you plan to stay on the same major version of an LTS for long, but I don't think it applies here.

Reply:
I personally disagree that it really indicates _how out-of-date_ a package is. For a package we have last publish date on the hackage. A very fresh package could use pretty old snapshot. Version of LTS isn't just a number, major version strongly attached to specific version of GHC, just a date in the name of a snapshot tells almost nothing about GHC version it uses (if you personally don't remember all the date checkpoints of releasing new LTS-es). Packages in a snapshot also could be very outdated of that date, they just have to be compatible with other dependent packages. Anyway, since you standing strong on your opinion about this, and nothing of it is changed it, feel free to close this PR because the purpose of it is totally opposite.